### ISSUE TITLE
Potential new Mechanics

### ISSUE THREAD

**Issue by Byteron** [wesnoth/haldric#29]
I'll leave this issue here as collection of features and new mechanics that come up during development.

So far we have
- inventories for units
- UtBS like AMLAs.
- terrain abilities / traps
~- area attacks~
~- true ranged attacks~
~- stackable units~

**Comment by GregoryLundberg** [comment]
Area and ranged attacks seem too unbalancing. But my opinion might change with more discussion.

I would suggest attacks be assigned movement points.  So I have a unit with 8 points, I could move for 2, attack for 4 and move for 2 (perhaps back to where I started); or I could attack for 4 and move for 4; etc.

**Comment by Elvish-Hunter** [comment]
> Area and ranged attacks seem too unbalancing.
Indeed they are if they aren't used carefully, that's why I suggested that core units shouldn't use real ranged attacks. However, I'd like to have support for them mainly for use by special units (my to-go example is the Cannon in Fate of a Princess), which can be part of a UMC, or even of a mainline campaign.

**Comment by mathpeseg** [comment]
I think that balancing is subjective, area attack is not bound to have a lot of damage, and enemies also could have units with area atacks. so i think this depends on how they are made. in my opinion this is a good thing to add because this game is aimed to be a somewhat of a sequel, and a new game with few changes is not something that will attract attention of new players. Most of them would think that the sequel is not even needed if there is not enough changes compared with the first game. 
Also, is too early to talk about balancing in a game that is not even defined yet.

**Comment by LawnCable** [comment]
> stackable units
I currently can't wrap my head around this one. How to display it, how many can be there at once?
Whats the limit(how many units can be in the same tile at once)?
For special tiles like villages or a cave entry I would like to have this feature. (though that would have big game-play/balance implications)

**Comment by Byteron** [comment]
Yeah, I don't think we will do stackable units.
same goes for true ranged attacks and area attacks.
I crossed what has too much of an impact on balancing.
Traps and an Inventory system still seem to be a good idea to me.

We might also want to rework AMLA. (Something like in UtBS, as an example)

**Comment by spartanatreyu** [comment]
> stackable units

I could see stackable units kind of being a thing if there was a Carry unit ability. There could be a transport ship unit that another unit could stack on top of and the transport ship could "carry" that unit over deep water.

The transport ship unit would 

- have health, movement, xp and traits like a regular unit.
- have the mechanical trait by default
- be able to move on deep water, shallow water, reefs (with a movement penalty)
- only be able to be recruited/recalled on villages in the above mentioned terrain types
- be able to be repaired when remaining still for a turn or when over a village at the start of a turn
- be able to be the movement target of a non-flying unit if it is not carrying a unit
- be able to drop-off a carried unit onto a terrain type it could normally move upon
- use the combat stats of the unit it is carrying in combat
- have no combat stats when not carrying a unit
- be able to be targeted independently of the unit it is carrying
- expand the gameplay meta (faster transport along constrained routes with the potential cost of losing your carried unit if the ship is destroyed on a terrain type that the carried unit cannot independently move upon)
- allow for some new interesting campaigns and water based maps (pirate, island-hopping and colonisation scenarios)
- cause the "carry" unit ability to be implemented into the game itself

Implementing the carry ability would allow add-on makers to
- create new custom transport units like an airship/eagle/floating-island/flying-carpet/gryphon/giant-moth/giant-turtle that carries units over certain terrain types (chasm/abyss/etc...)
- interesting neutral monsters like a dinosaur/dragon/giant-spider/kraken/etc... that carries an enemy unit back to its nest

I'm fairly sure a transport ship graphic already exists in the game as a kind of psudo-unit.

**Comment by spartanatreyu** [comment]
Besides the Carry unit ability, a Pack-Tactics unit ability could be interesting.

Grants a 20% damage buff when attacking a unit that was attacked by another unit of the same type as the currently attacking unit this turn. Would be useful for units like thieves, saurians, vampire bats, or any other weak + fast units

**Comment by spartanatreyu** [comment]
Random weather events that can impart damage or movement buffs/debuffs.

Clear - standard gameplay
Rain - Reduced movement, offence debuff to fire damage, defence buff to amphibious units (merfolk, naga, aquatic monsters)
Fog - Reduces line-of-sight in fog of war games, offence buff to ambush units
Dust storm - reduced movement, reduced line-of-sight in fog of war games
Snow - reduced movement, offensive buff to cold damage

Weather events must be somewhat predictable by map type (e.g. snow fall should not occur on swamp themed maps).

Weather events be should be forecasted at least one turn in advance and should end between the two player's turns that it started on to prevent a player getting an unfair advantage/disadvantage when chasing a unit. (e.g. Players A, B, C & D are playing. A weather event starts after player B's turn ends and before player C's turn begins. The weather event cannot end between player C & D, D & A or A & B's turns.)

**Comment by Byteron** [comment]
I do like the idea of natively supporting carry units.

"Pack" abilities exist in Wesnoth 1.x's UMC. It could make certain weaker units more interesting to play, but that affects balancing, thus might not be a suitable change.
At least not right away.

Weather is interesting (I thought about it as well) but definitely has a humongous impact on balancing.

**Comment by jay20162016** [comment]
For area/ranged attacks, maybe you could only allow it for siege weapons, thus making balancing potentially easier.

**Comment by jswise** [comment]
The lack of true ranged attacks in Wesnoth 1 has always bugged me. It's the one thing I would most like to change.

Yes, adding them would have a huge impact on balancing and any existing campaigns we want to import. However, I think it would be a big enough improvement to help justify the sequel's existence.

**Comment by DuzaBF** [comment]
Another new mechanic might be the addition of the multi-tile units. Large monsters like dragons or sea serpent can occupy 2 or 3 tiles.

**Comment by spartanatreyu** [comment]
> Another new mechanic might be the addition of the multi-tile units. Large monsters like dragons or sea serpent can occupy 2 or 3 tiles.

3 tiles might be easier than 2 shape-wise, but might seem weird in terms of scale (of course, this depends on the map behind the unit). I'm imagining an enormous stone titan the size of a large mountain lumbering across the map. It'd probably move very slow across the map, probably only a single tile at a time, and possibly even telegraphing which direction it's going to move in the next turn.

I could definitely see it getting used as boss units in campaigns. Large titan aside, I could also see it getting used for other things too: a swarm of locusts, a tornado, a magical maelstrom of interdimensional lightning, an evacuated town's population.

If large multi-tile units are added, that's going to add a lot of questions about pathing. What would such a unit do if they are moving through a mountain, city and forest at the same time, what speed bonuses are they given?

**Comment by ForestDragon-wesnoth** [comment]
> Indeed they are if they aren't used carefully, that's why I suggested that core units shouldn't use real ranged attacks. However, I'd like to have support for them mainly for use by special units (my to-go example is the Cannon in Fate of a Princess), which can be part of a UMC, or even of a mainline campaign.

yeah, such features would be really useful for UMC creators, and you'd have a rather hard time convincing people to switch to Project Haldric from original wesnoth unless you have huge new features like this.
